lib/fog/libvirt/requests/compute/list_domains.rb

```diff
@@ -46,6 +46,24 @@ def boot_order xml
           xml_elements(xml, "domain/os/boot", "dev")
         end
 
+        def firmware(xml)
+          firmware_from_loader = xml_elements(xml, "domain/os/loader", "type").first
+
+          case firmware_from_loader
+          when 'pflash'
+            'efi'
+          when 'rom'
+            'bios'
+          else
+            xml_elements(xml, "domain/os", "firmware").first || 'bios'
```

Comment:
Untested, but should this be
```suggestion
            xml_element(xml, "domain/os", "firmware") || 'bios'
```